### PR TITLE
Feature/blog improvements

### PR DESCRIPTION
**Pull request by just-at-uber**
### Added
- theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout. (see https://github.com/vuejs/vuepress/tree/master/packages/%40vuepress/theme-default/components)
- theme layout has been copied from vuepress-theme-blog and made small adjustments to work with default theme components. (see https://github.com/vuepress/vuepress-theme-blog/blob/master/layouts/GlobalLayout.vue)
- theme util has been copied from vuepress-theme-default which is used by those components. (see https://github.com/vuejs/vuepress/blob/master/packages/%40vuepress/theme-default/util/index.js)

### Changed
- Moved common config to /shared folder
- Moved common styling to /shared folder
- Navbar to share same component as default-theme
- Navbar links correctly navigate to non-blog site
- Small tweaks to ensure nav is working in blog theme

#### After
Screen Shot 2021-10-01 at 3 29 21 PM
Screen Shot 2021-10-01 at 3 29 31 PM
Screen Shot 2021-10-01 at 3 29 43 PM
Screen Shot 2021-10-01 at 3 29 52 PM

#### Before
135548778-c37feefc-9dba-425d-a2ce-0b46f9553709
135548784-f555a46f-7719-482a-8451-61825280b7db

### CODE REVIEW

File: src/blog/.vuepress/enhanceApp.js [hunk 1]
```diff
@@ -12,10 +12,10 @@ export default ({
 }) => {
```
Comment (just-at-uber) [inline src/blog/.vuepress/enhanceApp.js:12]: > Navbar links correctly navigate to non-blog site

File: src/shared/config.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+const { slackUrl } = require('./constants');
```
Comment (just-at-uber) [inline src/shared/config.js:1]: > Moved common config to /shared folder

File: src/shared/styles/index.styl [hunk 3]
```diff
@@ -0,0 +1,7 @@
+@require './code';
```
Comment (just-at-uber) [inline src/shared/styles/index.styl:1]: > Moved common styling to /shared folder

File: src/shared/styles/palette.styl [hunk 4]
```diff
@@ -0,0 +1,3 @@
+$uberBlue = #276ef1;
```
Comment (just-at-uber) [inline src/shared/styles/palette.styl:1]: > Moved common styling to /shared folder

File: src/shared/styles/nav.styl [hunk 5]
```diff
@@ -56,6 +56,7 @@
   .nav-dropdown {
```
Comment (just-at-uber) [inline src/shared/styles/nav.styl:56]: > small tweaks to ensure nav is working in blog theme

File: src/blog/.vuepress/theme/util/index.js [hunk 6]
```diff
@@ -0,0 +1,244 @@
+export const hashRE = /#.*$/
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/util/index.js:1]: > theme util has been copied from vuepress-theme-default which is used by those components.

File: src/.vuepress/components/slack-link.vue [hunk 7]
```diff
@@ -7,7 +7,7 @@
 </template>
 
 <script>
-import { slackUrl } from '../constants';
+import { slackUrl } from '../../shared/constants';
```
Comment (just-at-uber) [inline src/.vuepress/components/slack-link.vue:10]: > Moved common config to /shared folder

File: src/.vuepress/config.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-const { slackUrl } = require('./constants');
+const { themeConfig } = require('../shared/config');
```
Comment (just-at-uber) [inline src/.vuepress/config.js:1]: > Moved common config to /shared folder

File: src/.vuepress/styles/index.styl [hunk 9]
```diff
@@ -1,7 +1 @@
-@require './code';
-@require './global';
-@require './glossary';
-@require './grid';
-@require './home';
-@require './nav';
-@require './sidebar';
+@require '../../shared/styles';
```
Comment (just-at-uber) [inline src/.vuepress/styles/index.styl:1]: > Moved common styling to /shared folder

File: src/.vuepress/styles/palette.styl [hunk 10]
```diff
@@ -1,3 +1 @@
-$uberBlue = #276ef1;
-$readingProgressColor = $uberBlue;
-$navColor = rgba(25,29,46,1);
+@require '../../shared/styles/palette';
```
Comment (just-at-uber) [inline src/.vuepress/styles/palette.styl:1]: > Moved common styling to /shared folder

File: src/blog/.vuepress/theme/components/DropdownLink.vue [hunk 11]
```diff
@@ -0,0 +1,252 @@
+<template>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/components/DropdownLink.vue:1]: > theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout.

File: src/blog/.vuepress/theme/components/DropdownTransition.vue [hunk 12]
```diff
@@ -0,0 +1,33 @@
+<template>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/components/DropdownTransition.vue:1]: > theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout.

File: src/blog/.vuepress/theme/components/NavLink.vue [hunk 13]
```diff
@@ -0,0 +1,90 @@
+<template>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/components/NavLink.vue:1]: > > theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout.

File: src/blog/.vuepress/theme/components/NavLinks.vue [hunk 14]
```diff
@@ -0,0 +1,156 @@
+<template>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/components/NavLinks.vue:1]: > theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout.

File: src/blog/.vuepress/theme/components/Navbar.vue [hunk 15]
```diff
@@ -0,0 +1,131 @@
+<template>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/components/Navbar.vue:1]: > theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout.

File: src/blog/.vuepress/theme/components/Sidebar.vue [hunk 16]
```diff
@@ -0,0 +1,63 @@
+<template>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/components/Sidebar.vue:1]: > theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout.

File: src/blog/.vuepress/theme/components/SidebarButton.vue [hunk 17]
```diff
@@ -0,0 +1,40 @@
+<template>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/components/SidebarButton.vue:1]: > theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout.

File: src/blog/.vuepress/theme/components/SidebarGroup.vue [hunk 18]
```diff
@@ -0,0 +1,141 @@
+<template>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/components/SidebarGroup.vue:1]: > theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout.

File: src/blog/.vuepress/theme/components/SidebarLink.vue [hunk 19]
```diff
@@ -0,0 +1,133 @@
+<script>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/components/SidebarLink.vue:1]: > theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout.

File: src/blog/.vuepress/theme/components/SidebarLinks.vue [hunk 20]
```diff
@@ -0,0 +1,103 @@
+<template>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/components/SidebarLinks.vue:1]: > theme components have been copied from vuepress-theme-default and made small adjustments to work in blog layout.

File: src/blog/.vuepress/theme/layouts/GlobalLayout.vue [hunk 21]
```diff
@@ -0,0 +1,53 @@
+<template>
```
Comment (just-at-uber) [inline src/blog/.vuepress/theme/layouts/GlobalLayout.vue:1]: > theme layout has been copied from vuepress-theme-blog and made small adjustments to work with default theme components.

**Review by Groxx — APPROVED** [review]
Can't say I follow all that, but nothing suspicious lookin / yay, end result looks much improved!

**Pull request merged**